src/csharp/Microsoft.Spark/ML/Feature/FeatureHasher.cs

```diff
@@ -0,0 +1,145 @@
+            ((string[])_jvmObject.Invoke("getCategoricalCols")).ToList();
+        
+        /// <summary>
+        /// Marks columns as categorical columns.
+        /// </summary>
+        /// <param name="value">List of column names to mark as categorical columns</param>
+        /// <returns>New <see cref="FeatureHasher"/> object</returns>
+        public FeatureHasher SetCategoricalCols(IEnumerable<string> value) => 
+            WrapAsFeatureHasher(_jvmObject.Invoke("setCategoricalCols", value));
+        
+        /// <summary>
+        /// Gets the columns that the <see cref="FeatureHasher"/> should read from and convert into
+        /// hashes. This would have been set by SetInputCol.
+        /// </summary>
+        /// <returns>IEnumerable&lt;string&gt;, the input columns</returns>
+        public IEnumerable<string> GetInputCols() => (string[])_jvmObject.Invoke("getInputCols");
+
+        /// <summary>
+        /// Sets the columns that the <see cref="FeatureHasher"/> should read from and convert into
+        /// hashes.
+        /// </summary>
+        /// <param name="value">The name of the column to as use the source of the hash</param>
+        /// <returns>New <see cref="FeatureHasher"/> object</returns>
+        public FeatureHasher SetInputCols(IEnumerable<string> value) => 
+            WrapAsFeatureHasher(_jvmObject.Invoke("setInputCols", value));
+        
+        /// <summary>
+        /// Gets the number of features that should be used. Since a simple modulo is used to
+        /// transform the hash function to a column index, it is advisable to use a power of two
+        /// as the numFeatures parameter; otherwise the features will not be mapped evenly to the
+        /// columns.
+        /// </summary>
+        /// <returns>The number of features to be used</returns>
+        public int GetNumFeatures() => (int)_jvmObject.Invoke("getNumFeatures");
+
+        /// <summary>
+        /// Sets the number of features that should be used. Since a simple modulo is used to
+        /// transform the hash function to a column index, it is advisable to use a power of two as
+        /// the numFeatures parameter; otherwise the features will not be mapped evenly to the
+        /// columns.
+        /// </summary>
+        /// <param name="value">int</param>
+        /// <returns>New <see cref="FeatureHasher"/> object</returns>
+        public FeatureHasher SetNumFeatures(int value) => 
+            WrapAsFeatureHasher(_jvmObject.Invoke("setNumFeatures", value));
+        
+        /// <summary>
+        /// Gets the name of the column the output data will be written to. This is set by
+        /// SetInputCol.
+        /// </summary>
+        /// <returns>string, the output column</returns>
+        public string GetOutputCol() => (string)_jvmObject.Invoke("getOutputCol");
+
+        /// <summary>
+        /// The <see cref="FeatureHasher"/> will create a new column in the DataFrame, this is the
```

Comment:
This wording is a little confusing to me. Can we make it like, "Sets the name of the new column in the DataFrame created by <see cref="FeatureHasher">".. or something similar?